### ISSUE TITLE
iterable.length has been deprecated, use iterable.size or iterable.count()

### ISSUE THREAD

**Issue by mockdeep** [facebook/immutable-js#535]
I'm not sure if this is an issue with `immutable` or `poltergeist`, but I'm seeing the below error when I require `immutable` when running my tests with `poltergeist`. I don't see it when manually testing in Chrome or Firefox, though...

``` js
console.log('requiring immutable');
var Immutable = require('immutable');
console.log('required immutable');
```

```
requiring immutable
iterable.length has been deprecated, use iterable.size or iterable.count(). This warning will become a silent error in a future version. Error: 
    at http://127.0.0.1:36498/assets/application.js:8357
iterable.length has been deprecated, use iterable.size or iterable.count(). This warning will become a silent error in a future version. Error: 
    at http://127.0.0.1:36498/assets/application.js:8357
iterable.length has been deprecated, use iterable.size or iterable.count(). This warning will become a silent error in a future version. Error: 
    at http://127.0.0.1:36498/assets/application.js:8357
iterable.length has been deprecated, use iterable.size or iterable.count(). This warning will become a silent error in a future version. Error: 
    at http://127.0.0.1:36498/assets/application.js:8357
iterable.length has been deprecated, use iterable.size or iterable.count(). This warning will become a silent error in a future version. Error: 
    at http://127.0.0.1:36498/assets/application.js:8357
iterable.length has been deprecated, use iterable.size or iterable.count(). This warning will become a silent error in a future version. Error: 
    at http://127.0.0.1:36498/assets/application.js:8357
required immutable
```

**Comment by sonjapeterson** [comment]
Seeing the same issue here! @mockdeep did you ever find out what was causing this?

**Comment by mockdeep** [comment]
I suspect poltergeist is calling a `.length` on an object to figure out if it is iterable.

**Comment by born2net** [comment]
+1
any resolution to this?
can the error be removed?

**Comment by tshinnic** [comment]
In my case I do this horrible terrible sin:
```
    console.log('    props:   ', this.props)
```
Perfidious, isn't it?

This is a React app and I'm logging the latest props values upon entering a method.  Within those values in places may be one or three Immutable objects.  The browser's console.log (that disreputable sinner) is trying to determine what to do with this value - how to display it.  Apparently it treads upon the sore spot of `.length()` and Immutable vomits.

At least every 10 console log entries generates another full screen of "This warning will become a silent error" in some later generation.  This deprecation warning - on *every* single use and crushed toe - was very badly done.  It should have been generated only once, as person after person has mentioned.  

Not a good way to encourage adoption of your way of doing things, to harass users until they bend to your will?

Oh, look, version 3.8.0 is supposed to have gotten rid of this incessant warning.  I think I'll upgrade.  One can hope...   @#*&!! Wonder where the old copy is cached and getting reused...

BTW: I suppose you can close this report as fixed in 3.8.0 by https://github.com/facebook/immutable-js/commit/88f880fc02a377eb36dc1b9f3d6c86c2d1c287dd  Hmmm?

**Comment by amerdiglisic** [comment]
I am seeing it on 3.8.1 still.
